otelcol/command.go

```diff
@@ -12,6 +12,14 @@ import (
 	"go.opentelemetry.io/collector/featuregate"
 )
 
+// NewCommandFeatureGate constructs a new cobra.Command used to parse given FeatureGates.
+func NewCommandFeatureGate() *cobra.Command {
+	flagSet := flags(featuregate.GlobalRegistry())
```

Comment:
I guess here we could add something like `flagFeatureGate()` instead and add `FParseErrWhitelist: cobra.FParseErrWhitelist{UnknownFlags: false}` to the root command and add a No-op FeatureGate flag to the actual root command? Or simply parse them twice. Wdyt?